### PR TITLE
More crash fixes from Play logs..

### PR DESCRIPTION
**Pull request by hrydgard**
Fixes a real vertex decoder bug.

### CODE REVIEW

File: ext/native/ui/viewgroup.cpp [hunk 1]
```diff
@@ -1314,8 +1314,10 @@ EventReturn ListView::OnItemCallback(int num, EventParams &e) {
 	View *focused = GetFocusedView();
 	OnChoice.Trigger(ev);
 	CreateAllItems();
-	if (focused)
-		SetFocusedView(e.v);
+	// TODO: At this point, focused may no longer exist, depending on what OnChoice.Trigger does.
+	// Disable the refocus feature for now.
+	// if (focused)
+	//	SetFocusedView(e.v);
```
Comment (unknownbrackets) [inline ext/native/ui/viewgroup.cpp]: Hmm, this was introduced in de3c4.  Maybe we can use...

```c++
if (focused) {
	int offsetNum = num;
	for (int i = 0; i <= num; ++i) {
		if (hidden_.find(i) != hidden_.end()) {
			--offsetNum;
		}
	}

	SetFocusedView(linLayout_->GetViewByIndex(offsetNum));
```

-[Unknown]

Comment (hrydgard) [reply ext/native/ui/viewgroup.cpp]: Actually, things seem to still work without it, at least in the cases I tried (change buffered rendered mode, etc). So for 1.5.2 I'm leaving it like this.

File: GPU/Common/VertexDecoderArm64.cpp [hunk 2]
```diff
@@ -698,7 +698,12 @@ void VertexDecoderJitCache::Jit_PosS16Through() {
 }
 
 void VertexDecoderJitCache::Jit_NormalS8() {
-	LDRH(INDEX_UNSIGNED, tempReg1, srcReg, dec_->nrmoff);
+	// nrmoff can be odd in case of byte-only vertices! odd unsigned offsets are not allowed for LDRH.
+	// Switching to LDRB.
+	// Only seen this in a crash log.
+	LDRB(INDEX_UNSIGNED, tempReg1, srcReg, dec_->nrmoff);
+	LDRB(INDEX_UNSIGNED, tempReg3, srcReg, dec_->nrmoff + 1);
+	ORR(tempReg1, tempReg1, tempReg3, ArithOption(tempReg3, ST_LSL, 8));
```
[the remark this answers is not in the document]
Comment (hrydgard) [reply GPU/Common/VertexDecoderArm64.cpp]: Oh yeah, that's a better option.

File: ext/native/ui/viewgroup.cpp (hunk 1 above)
Comment (unknownbrackets) [reply ext/native/ui/viewgroup.cpp]: Ah, in that case let's just remove it.  I suppose we never use a ListView that doesn't close on select?

-[Unknown]

**Pull request merged**